director/director.go

```diff
@@ -148,11 +159,61 @@ func queryOrigins(ctx *gin.Context) {
 	ctx.JSON(http.StatusOK, res)
 }
 
+func handleFilterServer(ctx *gin.Context) {
```

Comment:
Can you add a comment header with the function's intended behavior and a quick note about where this gin context is usually coming from?